test/envoye2e/stackdriver_plugin/stackdriver_plugin_test.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client_test
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/golang/protobuf/jsonpb"
+	"istio.io/proxy/test/envoye2e/env"
+	fs "istio.io/proxy/test/envoye2e/stackdriver_plugin/fake_stackdriver"
+
+	"github.com/golang/protobuf/proto"
+	monitoringpb "google.golang.org/genproto/googleapis/monitoring/v3"
+)
+
+const outboundStackdriverFilter = `- name: envoy.wasm
+  config:
+    vm_config:
+      vm: "envoy.wasm.vm.null"
+      code:
+        inline_string: "envoy.wasm.metadata_exchange"
+    configuration: "test"
+- name: envoy.wasm
+  config:
+    vm_config:
+      vm: "envoy.wasm.vm.null"
+      code:
+        inline_string: "envoy.wasm.null.stackdriver"
+    configuration: >-
+      {
+        "kind": "OUTBOUND",
+        "monitoringEndpoint": "localhost:12312",
+      }`
+
+const inboundStackdriverFilter = `- name: envoy.wasm
+  config:
+    vm_config:
+      vm: "envoy.wasm.vm.null"
+      code:
+        inline_string: "envoy.wasm.metadata_exchange"
+    configuration: "test"
+- name: envoy.wasm
+  config:
+    vm_config:
+      vm: "envoy.wasm.vm.null"
+      code:
+        inline_string: "envoy.wasm.null.stackdriver"
+    configuration: >-
+      {
+        "kind": "INBOUND",
+        "monitoringEndpoint": "localhost:12312",
+      }`
+
+const outboundNodeMetadata = `istio.io/metadata: 
```

Comment:
before we add any more deps on `istio.io/metadata`, we need to resolve the issue of whether or not we are using that structure.

@kyessenov @mandarjog

Reply:
I will wait until we are settled on metadata stuff before pushing this.